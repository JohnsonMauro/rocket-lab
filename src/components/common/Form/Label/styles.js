import styled from '@emotion/styled';

const LabelStyle = styled.label`
  color: ${({ theme }) => theme.colors.blackRussian};
  display: inline-block;
  font-size: 13px;
  font-weight: 600;
  margin-bottom: 4px;
`;

export { LabelStyle };
