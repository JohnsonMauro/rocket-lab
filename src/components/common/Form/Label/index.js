import { LabelStyle } from './styles';

const Label = ({ className, children, forControl }) => (
  <LabelStyle className={className} htmlFor={forControl}>
    {children}
  </LabelStyle>
);

export default Label;
