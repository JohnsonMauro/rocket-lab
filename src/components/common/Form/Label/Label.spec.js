import React from 'react';
import { render } from '@testing-library/react';

import GlobalsContainer from '@test/GlobalsContainer';
import Label from './index';

describe('<Label />', () => {
  it('Renders the Label component', () => {
    const { container } = render(<Label forControl="sample">Sample</Label>, {
      wrapper: GlobalsContainer,
    });
    expect(container).toMatchSnapshot();
  });
});
