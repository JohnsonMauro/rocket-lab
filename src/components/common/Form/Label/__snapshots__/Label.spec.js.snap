// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Renders the Label component 1`] = `
<div>
  <label
    class="css-jdgriv-LabelStyle e6ushqh0"
    for="sample"
  >
    Sample
  </label>
</div>
`;
